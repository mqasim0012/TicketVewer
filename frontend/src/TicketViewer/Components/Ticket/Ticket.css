.ticket {
    background: lightgray;
    margin: 20px;
}