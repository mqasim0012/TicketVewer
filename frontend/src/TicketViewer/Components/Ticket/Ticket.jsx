import { React } from 'react'
import './Ticket.css'

const Ticket = (props) => {
    return (
        <div className='ticket'>
            {props.description}
        </div>
    );
}

export default Ticket;