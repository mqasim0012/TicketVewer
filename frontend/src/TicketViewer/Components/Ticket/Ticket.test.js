import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen, fireEvent } from '@testing-library/react'

import '@testing-library/jest-dom';
import Ticket from '.'

describe("Test ticket component behavior", () => {
    test('Ticket renders correctly', () => {
        const sampleTicket = renderer.create(
            <Ticket
                status="closed"
                subject="Need help"
                description="This feature does not work"
            />
        )

        const tree = sampleTicket.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test("Ticket button click test", () => {
        render(<Ticket
                status="open"
                subject="sub"
                description="descr"
             />)
        expect(screen.getByRole('button')).toHaveTextContent('View Details');

        fireEvent.click(screen.getByRole("button"));
        
        expect(screen.getByRole('button')).toHaveTextContent('Go back to all tickets');
    })

})