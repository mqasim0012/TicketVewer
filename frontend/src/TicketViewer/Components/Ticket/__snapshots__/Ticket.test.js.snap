// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ticket component behavior Ticket renders correctly 1`] = `
<div
  className="ticket "
>
  <div
    className="status"
  >
    Status: 
    closed
  </div>
  <div
    className="subject"
  >
    Subject: 
    Need help
  </div>
  <button
    className="view-more"
    onClick={[Function]}
  >
    View Details
  </button>
</div>
`;
